### PR TITLE
Adds capability to provide DisplayNameProvider for derived RecordType

### PR DESCRIPTION
**Pull request by jas-valgotar**
-Added Capability to pass DisplayNameProvider to RecordType via Constructor
-Added Test that passes Implementation of DisplayNameProvider to RecordType and verified it is working as expected.

### CODE REVIEW

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs [hunk 1]
```diff
@@ -18,6 +19,80 @@ namespace Microsoft.PowerFx.Interpreter.Tests
...
+            public override IEnumerable<string> FieldNames { get; }
+
+            public override bool TryGetFieldType(string name, out FormulaType type)
+            {
+                type = name switch
+                {
+                    "Num" => FormulaType.Number,
+                    "B" => FormulaType.Boolean,
+                    "Nested" => TableType.Empty().Add(new NamedFormulaType("Inner", FormulaType.Number, "InnerDisplay")),
+                    "Inner" => FormulaType.Number,
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs:33]: >         "Inner" => FormulaType.Number, [](http://example.com/codeflow?start=11&length=39)

Can we remove this?

Comment (jas-valgotar) [reply src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs]: #Resolved

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs [hunk 2]
```diff
@@ -18,6 +19,79 @@ namespace Microsoft.PowerFx.Interpreter.Tests
...
+                    _ => FormulaType.Blank
+                };
+
+                return type != FormulaType.Blank;
+            }
+
+            public LazyRecordType()
+                : base(new CustomDisplayNameProvider())
+            {
+                FieldNames = new List<string>() { "Num", "B", "Nested", "Inner" };
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs:42]: >"Inner" [](http://example.com/codeflow?start=72&length=7)

remove?

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Review by MikeStall — APPROVED** [review]
:shipit:

**Review by lesaltzm — APPROVED** [review]

File: src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs (hunk 2 above)
Comment (jas-valgotar) [reply src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs]: Removed.

**Pull request merged**